server/stream.go

```diff
@@ -1434,12 +1434,7 @@ func (mset *stream) setSourceConsumer(sname string, seq uint64) {
 			}
 			mset.mu.Unlock()
 		case <-time.After(10 * time.Second):
-			// Make sure things have not changed.
-			mset.mu.Lock()
-			if si := mset.sources[sname]; si != nil && si.cname == _EMPTY_ {
-				mset.setSourceConsumer(sname, seq)
-			}
-			mset.mu.Unlock()
+			return
```

Comment:
Still think that we should remove the retry that we do when getting a bad response (line 1429), and same for mirrors. Again, the long lived go routines that are checking for stalled consumers *should* be able to handle those cases.